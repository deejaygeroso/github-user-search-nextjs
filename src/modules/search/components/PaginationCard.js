import React from "react";
import PropTypes from "prop-types";
import CssBaseline from "@material-ui/core/CssBaseline";
import { createMuiTheme, MuiThemeProvider } from "@material-ui/core/styles";
import Pagination from "material-ui-flat-pagination";

const theme = createMuiTheme();
const PER_PAGE_LIMIT = 8;

const PaginationCard = ({ total, offset, onClick, isHidden }) => {
  // hide if loading is showing
  if(isHidden){
    return <div/>
  }
  // do not show pagination
  if (total <= PER_PAGE_LIMIT) {
    return <div />;
  }
  return (
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      <Pagination
        limit={20}
        offset={offset}
        total={total}
        onClick={(e, offset) => {
          let page = offset / PER_PAGE_LIMIT + 1;
          onClick(page);
        }}
      />
    </MuiThemeProvider>
  );
};

PaginationCard.propTypes = {
  isHidden: PropTypes.bool.isRequired,
  offset: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired
};

export default PaginationCard;
