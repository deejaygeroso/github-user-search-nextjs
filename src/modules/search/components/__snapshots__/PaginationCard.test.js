import React from "react";
import PaginationCard from "../PaginationCard";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const props = {
    total: 10,
    offset: 20,
    onClick: () => {},
    isHidden: false
  };
  const tree = renderer.create(<PaginationCard {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
