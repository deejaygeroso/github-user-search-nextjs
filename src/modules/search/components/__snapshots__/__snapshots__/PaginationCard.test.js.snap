// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFlatPagination-root-1"
>
  <button
    className="MuiButtonBase-root-75 MuiButton-root-49 MuiFlatPageButton-root-25 MuiFlatPageButton-rootEnd-28 MuiFlatPagination-rootEnd-4 MuiButton-text-51 MuiFlatPageButton-text-31 MuiFlatPagination-text-7 MuiButton-textPrimary-52 MuiFlatPageButton-textPrimary-32 MuiFlatPagination-textPrimary-8 MuiButton-flat-54 MuiButton-flatPrimary-55"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-50 MuiFlatPageButton-label-30 MuiFlatPagination-label-6"
    >
      &lt;
    </span>
    <span
      className="MuiTouchRipple-root-78"
    />
  </button>
  <button
    className="MuiButtonBase-root-75 MuiButton-root-49 MuiFlatPageButton-root-25 MuiFlatPageButton-rootStandard-29 MuiFlatPagination-rootStandard-5 MuiButton-text-51 MuiFlatPageButton-text-31 MuiFlatPagination-text-7 MuiButton-textPrimary-52 MuiFlatPageButton-textPrimary-32 MuiFlatPagination-textPrimary-8 MuiButton-flat-54 MuiButton-flatPrimary-55"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-50 MuiFlatPageButton-label-30 MuiFlatPagination-label-6"
    >
      1
    </span>
    <span
      className="MuiTouchRipple-root-78"
    />
  </button>
  <button
    className="MuiButtonBase-root-75 MuiButton-root-49 MuiFlatPageButton-root-25 MuiFlatPageButton-rootStandard-29 MuiFlatPagination-rootStandard-5 MuiButton-text-51 MuiFlatPageButton-text-31 MuiFlatPagination-text-7 MuiButton-textPrimary-52 MuiFlatPageButton-textPrimary-32 MuiFlatPagination-textPrimary-8 MuiButton-flat-54 MuiButton-flatPrimary-55"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-50 MuiFlatPageButton-label-30 MuiFlatPagination-label-6"
    >
      2
    </span>
    <span
      className="MuiTouchRipple-root-78"
    />
  </button>
  <button
    className="MuiButtonBase-root-75 MuiButton-root-49 MuiFlatPageButton-root-25 MuiFlatPageButton-rootCurrent-26 MuiFlatPagination-rootCurrent-2 MuiButton-text-51 MuiFlatPageButton-text-31 MuiFlatPagination-text-7 MuiButton-textSecondary-53 MuiFlatPageButton-textSecondary-33 MuiFlatPagination-textSecondary-9 MuiButton-flat-54 MuiButton-flatSecondary-56"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-50 MuiFlatPageButton-label-30 MuiFlatPagination-label-6"
    >
      3
    </span>
  </button>
  <button
    className="MuiButtonBase-root-75 MuiButtonBase-disabled-76 MuiButton-root-49 MuiFlatPageButton-root-25 MuiFlatPageButton-rootEnd-28 MuiFlatPagination-rootEnd-4 MuiButton-text-51 MuiFlatPageButton-text-31 MuiFlatPagination-text-7 MuiButton-textPrimary-52 MuiFlatPageButton-textPrimary-32 MuiFlatPagination-textPrimary-8 MuiButton-flat-54 MuiButton-flatPrimary-55 MuiButton-disabled-69 MuiFlatPageButton-disabled-37 MuiFlatPagination-disabled-13"
    disabled={true}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="button"
  >
    <span
      className="MuiButton-label-50 MuiFlatPageButton-label-30 MuiFlatPagination-label-6"
    >
      &gt;
    </span>
  </button>
</div>
`;
